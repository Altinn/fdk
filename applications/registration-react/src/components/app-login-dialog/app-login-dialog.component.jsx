import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

import localization from '../../lib/localization';
import { login } from '../../auth/auth-service';
import './app-login-dialog.scss';

export const LoginDialogPure = ({ loggedOut }) => {
  return (
    <div className="login-dialog-wrapper p-5">
      <div className="container">
        <div className="row">
          <div className="col-12">
            <div className="col-12 col-md-8">
              {loggedOut && (
                <div
                  className="mt-2 alert alert-warning fdk-text-size-small fdk-color-neutral-darkest"
                  role="alert"
                >
                  <span>
                    <strong>
                      {localization.loginDialog.loggedOutMsgPart1}
                    </strong>
                  </span>
                  <span>{localization.loginDialog.loggedOutMsgPart2}</span>
                </div>
              )}
              {!loggedOut && (
                <div>
                  <h1 className="fdk-text-extra-strong mb-md-5">
                    {localization.loginDialog.title}
                  </h1>
                  <div className="fdk-text-size-medium fdk-text-line-medium mt-2 mb-md-3">
                    {localization.loginDialog.ingress}
                  </div>
                </div>
              )}
              <div className="mt-5 mb-5">
                <Button className="fdk-button fdk-button-cta" onClick={login}>
                  {localization.app.logIn}
                </Button>
              </div>
              <div className="fdk-text-size-small fdk-text-line-medium">
                <strong>
                  {localization.catalogs.missingCatalogs.accessTitle}
                </strong>
                <p>
                  <a href="https://fellesdatakatalog.brreg.no/about-registration">
                    {localization.catalogs.missingCatalogs.accessText}
                    <i className="fa fa-external-link fdk-fa-right" />
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

LoginDialogPure.defaultProps = {
  loggedOut: false
};

LoginDialogPure.propTypes = {
  loggedOut: PropTypes.bool
};

function mapStateToProps({ user }) {
  const { userItem, isFetchingUser } = user || {
    userItem: null,
    isFetchingUser: false
  };
  return {
    userItem,
    isFetchingUser
  };
}

export const LoginDialog = connect(mapStateToProps)(LoginDialogPure);
