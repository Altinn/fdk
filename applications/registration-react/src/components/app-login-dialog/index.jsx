import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import axios from 'axios';

import localization from '../../utils/localization';
import './index.scss';

//const LoginDialog = props => {
class LoginDialog extends React.Component {
  componentWillMount() {
    if(this.props.loggedOut) {
      return axios.get('/logout')
        .then((response) => response)
        .catch((response) => {
          const { error } = response;
          return Promise.reject(error);
        })

      return axios.get(
        '/logout'
      ).then((response) => {
      }).catch((error) => {
        throw {error}
      });
    }
  }
  render() {
    const {loggedOut} = this.props;
    return (
      <div className="login-dialog-wrapper p-5">
        <div className="container-fluid">
          <div className="row">
            <div className="col-12 col-md-8 offset-md-2">
              <div className="col-12 col-md-8">
                {loggedOut &&
                (
                  <div className="mt-2 alert alert-warning fdk-text-size-small fdk-color1" role="alert">
                    <span>
                      <strong>{localization.loginDialog.loggedOutMsgPart1}</strong>
                    </span>
                    <span>
                      {localization.loginDialog.loggedOutMsgPart2}
                    </span>
                  </div>
                )
                }
                {!loggedOut &&
                (
                  <div>
                    <h1 className="fdk-text-extra-strong mb-md-5">{localization.loginDialog.title}</h1>
                    <div className="fdk-text-size-medium fdk-text-line-medium mt-2 mb-md-3">
                      {localization.loginDialog.ingress}
                    </div>
                  </div>
                )
                }
                <a className="fdk-button fdk-button-cta mb-2 mb-md-5" href="/login">
                  {localization.app.logIn}
                </a>
                <div className="fdk-text-size-small fdk-text-line-medium">
                  <strong>{localization.catalogs.missingCatalogs.accessTitle}</strong>
                  <p>
                    {localization.catalogs.missingCatalogs.accessText}
                  </p>
                  <strong>{localization.catalogs.missingCatalogs.assignAccessTitle}</strong>
                  <p>
                    {localization.catalogs.missingCatalogs.assignAccessText}
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default LoginDialog;
