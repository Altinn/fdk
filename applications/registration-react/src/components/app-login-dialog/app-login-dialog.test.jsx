import React from 'react';
import { shallow } from 'enzyme';
import { LoginDialogPure } from './app-login-dialog.component';

let defaultProps;
let wrapper;
let dispatch;

beforeEach(() => {
  dispatch = jest.fn();
  defaultProps = {
    dispatch,
    loggedOut: true
  };
  wrapper = shallow(<LoginDialogPure {...defaultProps} />);
});

test('should render LoginDialog correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
