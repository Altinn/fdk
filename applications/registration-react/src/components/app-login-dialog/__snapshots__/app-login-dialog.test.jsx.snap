// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginDialog correctly 1`] = `
<div
  className="login-dialog-wrapper p-5"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <div
          className="col-12 col-md-8"
        >
          <div
            className="mt-2 alert alert-warning fdk-text-size-small fdk-color-neutral-darkest"
            role="alert"
          >
            <span>
              <strong>
                Du har blitt logget ut. 
              </strong>
            </span>
            <span>
              Sesjonen ble tidsavbrutt, men vi har tatt vare på alle endringene du har gjort. Logg inn på nytt for å fortsette å jobbe med registreringen.
            </span>
          </div>
          <div
            className="mt-5 mb-5"
          >
            <Button
              className="fdk-button fdk-button-cta"
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              Logg inn
            </Button>
          </div>
          <div
            className="fdk-text-size-small fdk-text-line-medium"
          >
            <strong>
              Trenger du tilgang?
            </strong>
            <p>
              <a
                href="https://fellesdatakatalog.brreg.no/about-registration"
              >
                Les mer om hvordan du får tilgang her
                <i
                  className="fa fa-external-link fdk-fa-right"
                />
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
